### PR TITLE
[Alberto Icardi]- Algorithms task 1

### PR DESCRIPTION
**Pull request by AlbertoIcardi97**

### CODE REVIEW

File: algorithms_task_1/student/include/impl.h [hunk 1]
```diff
@@ -11,4 +11,12 @@
   * @return consecutive range
 */
 template<class FI, class Comparator = std::equal_to<typename std::iterator_traits<FI>::value_type>>
-std::pair<FI, FI> consecutive_group(FI start, FI end, Comparator comp = {});
+std::pair<FI, FI> consecutive_group(FI start, FI end, Comparator comp = {}){
+  auto first=std::adjacent_find(start,end, comp);
+  auto second=std::adjacent_find(first,end, std::not_equal_to<typename std::iterator_traits<FI>::value_type>());
```
Comment (smay1613) [inline algorithms_task_1/student/include/impl.h]: and what if comparator is not an std::equal?

File: algorithms_task_1/student/include/impl.h [hunk 2]
```diff
@@ -13,7 +13,12 @@
 template<class FI, class Comparator = std::equal_to<typename std::iterator_traits<FI>::value_type>>
 std::pair<FI, FI> consecutive_group(FI start, FI end, Comparator comp = {}){
   auto first=std::adjacent_find(start,end, comp);
-  auto second=std::adjacent_find(first,end, std::not_equal_to<typename std::iterator_traits<FI>::value_type>());
+
+  auto not_comp= [&](const typename std::iterator_traits<FI>::value_type & a,const typename std::iterator_traits<FI>::value_type &  b)
```
Comment (smay1613) [inline algorithms_task_1/student/include/impl.h:17]: also you may try std::not2 and std::not_fn

File: algorithms_task_1/student/include/impl.h [hunk 3]
```diff
@@ -13,7 +13,12 @@
...
+  {return !comp(a,b);};
+  // Since C++17 I could have used std::not_fn(comp) !!!
```
Comment (smay1613) [inline algorithms_task_1/student/include/impl.h:19]: you may try std::not2, but that's not good idea

Comment (AlbertoIcardi97) [reply algorithms_task_1/student/include/impl.h:19]: So, is my way correct?

Comment (smay1613) [reply algorithms_task_1/student/include/impl.h:19]: Yes!

**Pull request closed without merging**